coralproject/talk-fe
# Internationalization

## What does this PR do?

Enables internationalization across talk-fe
## How do I test this PR?
1. Start a server with the launch script in `talk`
2. Go to `packages/coral-embed-stream`
3. `npm run dev-start`
4. Buttons to post a comment, etc  should appear in English
5. Open a console and enter `localStorage.setItem(‘locale’, ‘es-US’)`
6. Reload the page. Buttons should appear in Spanish (with some missing translations.
7. Navigate to `packages/coral-admin`
8. `npm start`
9. Internationalization should also work in `coral-admin`

———————
## Notes

The internationalization logic is stored in `coral-framework`. It is initialized with a javascript object of the form: 

```
const translations = {
    en: {
           stuff: ‘things’,
           nested: {
                 moreStuff: “more things”
           },
     es: {
            //Repeticiones
     }
}
```

This creates an object which can be used to translate like so:

```
import {I18n} from ‘coral-framework’
const lang = new I18n(translations)
lang.t(‘stuff’) // things
lang.t(‘nested.morestuff’) //morethings
```

I18n will detect the user’s language from navigator.locale

Translation objects should be stored on a per-plugin basis to avoid unnecessary language loading.
